### ISSUE TITLE
Attack has been blocked by CORS policy

### ISSUE THREAD

**Issue by peace27-96** [nccgroup/singularity#44]
Hi @gdncc and congratulations on the tool you made. I am trying to perform the attack on my home router and I cannot understand why it is giving me this error.

![Screenshot from 2021-09-25 18-00-32](https://user-images.githubusercontent.com/26538947/134778983-8aed3121-a17f-4b34-ac43-1b380175f9a2.jpg)

I tried another router on a different line and the attack was successful.

Could you help me understand why it doesn't work on this router? or what caused the error.

Sorry for my English.

**Comment by sanktjodel** [comment]
From the screenshot, it looks like the DNS rebinding did not succeed. Did you try to manually verify that the target system is vulnerable to DNS rebinding as explained here https://github.com/nccgroup/singularity/wiki/Testing-for-Vulnerable-Services ?

**Comment by sanktjodel** [comment]
Thank you for the update. We believe this is the cause of a recent new feature in the Chrome browser described at https://developer.chrome.com/blog/private-network-access-update/. For now it looks like rebinding to a private IP address in Chrome does no longer work. We may explore this in the future.

**Comment by minanagehsalalma** [comment]
> Thank you for the update. We believe this is the cause of a recent new feature in the Chrome browser described at https://developer.chrome.com/blog/private-network-access-update/. For now it looks like rebinding to a private IP address in Chrome does no longer work. We may explore this in the future.

@sanktjodel  i think you may want to take a look on these two then
https://github.com/nccgroup/singularity/issues/36
https://github.com/nccgroup/singularity/issues/35

also adding an option for auto grabbing the public IP for using in the manager in case of chrome detected would be useful , till you happen to figure a fix.

Thanks ;)

**Comment by web3res** [comment]
Thanks , we await updates on the possible resolution, currently only Chrome is blocking it but in some cases it works

**Comment by sanktjodel** [comment]
We have documented this issue in our new blog post at https://research.nccgroup.com/2023/04/27/state-of-dns-rebinding-in-2023/ explaining Local Network Access, a new draft W3C specification, implemented in Chrome. The blog post includes two ways to bypass these restrictions with restricted scope.

We have also documented the error here: https://github.com/nccgroup/singularity/wiki/Common-Issues

Thanks for bringing up the matter. Closing this issue.
